lib/service/storage/elasticsearch.ts

```diff
@@ -3218,9 +3436,48 @@ class ElasticSearch extends Service {
       searchBody.query = { match_all: {} };
     }
 
+    if (index && this.virtualIndex.isVirtual(index)) {
+      this._sanitizeSearchBodyForVirtualIndex(searchBody, index);
+    }
+
     return searchBody;
   }
 
+  _sanitizeSearchBodyForVirtualIndex(searchBody, index) {
```

Comment:
I think this method should be tested unitary with as most as possible usecases. Also, it should be in the `VirtualIndex` service

Reply:
If I remember will, we said that we do not continue to create unit test. 
Is it ok if I test all possible usecase with functionnal test?

Reply:
In some case unit tests may be more easy to write and fast to execute. This is a good example because you would have to try many differents search requests as input and doing so with functionnal tests would take a lot of extra time:
 - having to send network request
 - having to play it 6 times in total: Node.js 14 and 16 with HTTP, WebSocket and MQTT